e2e/e2e_test.go

```diff
@@ -502,6 +502,87 @@ func TestE2E_ConfigStreamlining_Deprecations(t *testing.T) {
 	validateServices(t, true, []string{"api"}, resourcesPath)
 }
 
+// TestE2E_TriggerTaskWhenActive tests making changes to monitored dependencies
+// while a task is running. It verifies:
+//
+// 1. a change to the same task will trigger the task after the run completes
+// 2. a change to a different task runs immediately
+// 3. a change after the task completes will still trigger the task to run
+//
+// https://github.com/hashicorp/consul-terraform-sync/issues/732
```

Comment:
One other test case I tried manually, is making sure that the task that applies is the latest change i.e.

- register a service (Change 1)
- CTS starts up task
- deregister a service (Change 2)
- register a service (Change 3)
- CTS completes first task
- CTS executes a new task for Change 3 not 2

Do you think this is worth automating? I verified that it behaves as expected manually

Reply:
I also verified the buffering still works manually. One day we'll create those buffering tests ...

Reply:
Yeah, I think it would be good to test that the latest change is used! Thinking about whether it should be a separate test or added to this one, though! 🤔

Reply:
Added and went with a separate test!